introspection-engine/connectors/sql-introspection-connector/src/calculate_datamodel.rs

```diff
@@ -55,6 +60,50 @@ impl<'a> CalculateDatamodelContext<'a> {
             .map(|enm| enm.name())
             .unwrap_or_else(|| self.schema.walk(id).name())
     }
+
+    pub(crate) fn existing_inline_relation(&self, id: sql::ForeignKeyId) -> Option<walkers::InlineRelationWalker<'a>> {
+        self.existing_relations
+            .get(&id)
+            .map(|relation_id| self.previous_schema.db.walk(*relation_id).refine().as_inline().unwrap())
+    }
+
+    pub(crate) fn existing_m2m_relation(
+        &self,
+        id: sql::TableId,
+    ) -> Option<walkers::ImplicitManyToManyRelationWalker<'a>> {
+        self.existing_m2m_relations.get(&id).map(|relation_id| {
+            self.previous_schema
+                .db
+                .walk(*relation_id)
+                .refine()
+                .as_many_to_many()
+                .unwrap()
```

Comment:
This is safe because we only put many-to-many relations in that hashmap, but we can definitely make it safer/tighter. I'll do that.